### PR TITLE
tp: add SQLite VFS backed by abstract I/O

### PR DESCRIPTION
**Pull request by LalitMaganti**
Add a random-access FileSystem interface and a SQLite VFS backed by it.

Use process-wide local and no-op filesystem implementations, with reusable seek and truncate utilities.

### CODE REVIEW

File: src/base/file_utils.cc [hunk 1]
```diff
@@ -265,6 +266,36 @@ bool FlushFile(int fd) {
 #endif
 }
 
+bool SeekFile(int fd, uint64_t offset) {
+  if (offset > static_cast<uint64_t>(std::numeric_limits<int64_t>::max())) {
```
Comment (safayat-google) [inline src/base/file_utils.cc:271]: nit: check against `std::numeric_limits<off_t>::max()` for non-windows. also in `TruncateFile()`

File: src/base/file_utils.cc [hunk 2]
```diff
@@ -265,6 +266,36 @@ bool FlushFile(int fd) {
 #endif
 }
 
+bool SeekFile(int fd, uint64_t offset) {
```
Comment (safayat-google) [inline src/base/file_utils.cc:269]: nit: check for invalid fd?

File: src/base/file_utils.cc [hunk 3]
```diff
@@ -265,6 +266,36 @@ bool FlushFile(int fd) {
...
+  }
+#if PERFETTO_BUILDFLAG(PERFETTO_OS_WIN)
+  int result = _chsize_s(fd, size);
+  if (result != 0) {
+    errno = result;
+    return false;
+  }
+  return true;
+#else
+  return ftruncate(fd, static_cast<off_t>(size)) == 0;
```
Comment (safayat-google) [inline src/base/file_utils.cc]: nit: wrap with `PERFETTO_EINTR()`?

File: src/base/utils_unittest.cc [hunk 4]
```diff
@@ -364,6 +364,28 @@ TEST(UtilsTest, CopyFileContents) {
...
+  ASSERT_EQ(WriteAll(*file, "abcdef", 6), 6);
+
+  ASSERT_TRUE(TruncateFile(*file, 3));
+  EXPECT_EQ(GetFileSize(*file), 3u);
+
+  std::string contents;
+  ASSERT_TRUE(ReadFile(file.path(), &contents));
+  EXPECT_EQ(contents, "abc");
+}
+
```
Comment (safayat-google) [inline src/base/utils_unittest.cc:404]: nit and optional: add tests to cover with growing size(TruncateFile), invalid fd and overflow case?

**Review by safayat-google — APPROVED** [review]

File: src/base/file_utils.cc (hunk 2 above)
Comment (LalitMaganti) [reply src/base/file_utils.cc:269]: Leaving this to the syscall, which rejects invalid descriptors with EBADF.

File: src/base/file_utils.cc (hunk 3 above)
Comment (LalitMaganti) [reply src/base/file_utils.cc]: Wrapped ftruncate. lseek is not interruptible and does not report EINTR.

File: src/base/utils_unittest.cc (hunk 4 above)
Comment (LalitMaganti) [reply src/base/utils_unittest.cc:404]: Done; added growth, invalid-fd, and overflow coverage.

File: src/base/file_utils.cc (hunk 1 above)
Comment (LalitMaganti) [reply src/base/file_utils.cc:271]: Done. This matters on platforms or build configurations where off_t is narrower than 64 bits.
